Library/ViewModels/CommentsViewModel.swift

```diff
@@ -252,10 +260,21 @@ public final class CommentsViewModel: CommentsViewModelType,
       self.loadCommentsAndProjectIntoDataSource.ignoreValues()
     )
 
-    self.configureFooterViewWithState = Signal.merge(
+    let hideFooterView: Signal<CommentTableViewFooterViewState, Never> = Signal.merge(
       initialLoadOrReload.mapConst(.hidden),
+      errors
+        .mapConst(.hidden)
+        .withLatestFrom(hasRequestedNextPage.filter(isFalse))
+        .map(first)
+    )
+
+    self.configureFooterViewWithState = Signal.merge(
+      hideFooterView,
       footerViewActivityState.mapConst(.activity),
-      errors.mapConst(.error)
+      errors
+        .mapConst(.error)
+        .withLatestFrom(hasRequestedNextPage.filter(isTrue))
+        .map(first)
```

Comment:
I'm a bit confused by this logic, does it mean that only next page requests with an error hide the footer view?

Reply:
Add some inline docs here too.